bdd/features/instance-lifecycle.feature

```diff
@@ -0,0 +1,32 @@
+Feature: Instance lifecycle API behavior
+
+    Background:
+        Given host is running
+
+    Scenario: Starting an instance
+        Given sequence "hello" is available
+        When SequenceClient.start is executed
+        Then it should return an InstanceClient
+        And instance status should be "starting"
+        When instance starts producing output
+        Then instance status should be "running"
+
+    Scenario: Instance completes successfully
+        Given instance "hello" is running
+        When instance output is completed
+        Then instance status should immediately change to "finishing"
+        When wait until runner has ended execution
+        Then instance status should immediately change to "completed"
```

Comment:
I'm not sure about this. Do we actually need `finishing` status? Is there a real difference between output being completed (i.e. stream is ended and returned) and runner shutdown? How does runner exit delay affect this? I think we should discuss what @kociolekscramjet and @agne6ka expect here and whether backend can behave in such way.

Reply:
Runner has 10seconds exit delay. So when sequence is ended (stream ended/promise resolved, smth returned) and SEQUENCE_COMPLETED control message is sent there is 10secs of `sleep` in Runner before it exits. this is the `finishing` status. when runner returns exitCode we determine if instance is `errored` or `completed`